include/alpaka/acc/AccOmp5.hpp

```diff
@@ -15,6 +15,12 @@
 #        error If ALPAKA_ACC_ANY_BT_OMP5_ENABLED is set, the compiler has to support OpenMP 4.0 or higher!
 #    endif
 
+// Kill printf in AMD GPU code because of missing compiler support
+#    ifdef __AMDGCN__
```

Comment:
This is disabling `printf` for HIP too but I assume OpenMP offloading can not be enabled together with the HIP backend.

Reply:
Enabling both of these at the same time is not a use case I had considered. So, at the moment this would not be possible, or you have to go without `printf` in your HIP kernels, too.

Reply:
Maybe it would also work with:
```c++
#ifdef __AMDGCN__ && !defined __HIP__
```
This would assume, that the OpenMP HSA device code compiler does not define `__HIP__` even is HIP enabled otherwise.

This interaction is pure speculation though: We have no compiler that can properly offload OpenMP, let alone doe than *and* HIP. So, there would be no way to test this modification. It is likely, that this workaround becomes obsolete,, by `printf` becoming available in clang HSA before anyone tries to run OpenMP target and HIP together.